AbstractDevServerRunnerTest.shouldPassEncodedUrlToDevServer is flaky

AbstractDevServerRunnerTest.shouldPassEncodedUrlToDevServer randomly fails on GitHub actions validation

```
2023-11-29T12:49:25.3720681Z [ERROR] com.vaadin.base.devserver.AbstractDevServerRunnerTest.shouldPassEncodedUrlToDevServer -- Time elapsed: 0.555 s <<< FAILURE!
2023-11-29T12:49:25.3722902Z java.lang.AssertionError: Dev server should have served the resource
2023-11-29T12:49:25.3724342Z 	at org.junit.Assert.fail(Assert.java:89)
2023-11-29T12:49:25.3727476Z 	at org.junit.Assert.assertTrue(Assert.java:42)
2023-11-29T12:49:25.3729260Z 	at com.vaadin.base.devserver.AbstractDevServerRunnerTest.shouldPassEncodedUrlToDevServer(AbstractDevServerRunnerTest.java:120)
2023-11-29T12:49:25.3731437Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-11-29T12:49:25.3733353Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-11-29T12:49:25.3735463Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-11-29T12:49:25.3741126Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-11-29T12:49:25.3742807Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2023-11-29T12:49:25.3744814Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2023-11-29T12:49:25.3843190Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2023-11-29T12:49:25.3845271Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2023-11-29T12:49:25.3847357Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2023-11-29T12:49:25.3849453Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2023-11-29T12:49:25.3851290Z 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
2023-11-29T12:49:25.3852798Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-11-29T12:49:25.3854404Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2023-11-29T12:49:25.3856159Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2023-11-29T12:49:25.3857753Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2023-11-29T12:49:25.3859696Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2023-11-29T12:49:25.3861260Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2023-11-29T12:49:25.3862636Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2023-11-29T12:49:25.3864089Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2023-11-29T12:49:25.3866138Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2023-11-29T12:49:25.3867575Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2023-11-29T12:49:25.3869005Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-11-29T12:49:25.3870386Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2023-11-29T12:49:25.3871624Z 	at org.junit.runners.Suite.runChild(Suite.java:128)
2023-11-29T12:49:25.3872591Z 	at org.junit.runners.Suite.runChild(Suite.java:27)
2023-11-29T12:49:25.3873413Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2023-11-29T12:49:25.3875240Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2023-11-29T12:49:25.3876394Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2023-11-29T12:49:25.3877754Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2023-11-29T12:49:25.3879072Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2023-11-29T12:49:25.3880615Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2023-11-29T12:49:25.3881894Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2023-11-29T12:49:25.3883014Z 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:49)
2023-11-29T12:49:25.3884859Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:120)
2023-11-29T12:49:25.3887384Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:95)
2023-11-29T12:49:25.3889437Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
2023-11-29T12:49:25.3891222Z 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:69)
2023-11-29T12:49:25.3893001Z 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:146)
2023-11-29T12:49:25.3894987Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
2023-11-29T12:49:25.3896706Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2023-11-29T12:49:25.3898249Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
2023-11-29T12:49:25.3899731Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
```

--- a/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/AbstractDevServerRunnerTest.java
+++ b/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/AbstractDevServerRunnerTest.java
@@ -100,6 +100,7 @@ public class AbstractDevServerRunnerTest extends AbstractDevModeTest {
     @Test
     public void shouldPassEncodedUrlToDevServer() throws Exception {
         handler = new DummyRunner();
+        waitForDevServer();
         DevModeHandler devServer = Mockito.spy(handler);
 
         HttpServletResponse response = Mockito.mock(HttpServletResponse.class);

